tsdb/store.go

```diff
@@ -237,7 +237,11 @@ func (s *Store) loadShards() error {
 				}
 
 				shard := NewShard(s.databaseIndexes[db], path, s.EngineOptions)
-				shard.Open()
+				err = shard.Open()
+				if err != nil {
+					s.Logger.Printf("failed to open shard %d: %s", shardID, err)
```

Comment:
The check here https://github.com/DanielMorsing/influxdb/blob/sharderrcheck/tsdb/store.go#L233, is making sure the file in the directory is actually named how shards we created are named.  It's to prevent trying to open something like `1.bak` which is not a file we would create. 

L242 should return the error because we should only ever be attempting to open valid shards at this point.  A failure would be problem.